collectors/AlertCollector.py

```diff
@@ -73,7 +66,7 @@ def collect(self):
 
     def generate_alert_label_values(self, alert):
         alert_labels = dict()
-        alert_entry = self.alertdefinitions.get(alert.get('alertDefinitionId', {}), {})
+        alert_entry = self.get_alertdefinition(alert.get('alertDefinitionId', {}))
```

Comment:
This would add an additional call for each alert we're getting back from the `get_alerts` call.

It is a tradeoff: memory vs api load.

I'd rather suggest to implement a cache to keep the API load minimal and not have all of the definitions stored: 
1. look if there alertdefinition is locally available
2. take it, if it is not there, fetch it and take it